lib/bamboo/adapters/elastic_email_adapter.ex

```diff
@@ -197,10 +197,42 @@ defmodule Bamboo.ElasticEmailAdapter do
 
   defp put_transactional(map), do: Map.put(map, "isTransactional", true)
 
-  defp put_post_back(%{private: %{elastic_custom_vars: %{post_back: post_back}}} = map),
-    do: Map.put(map, "postBack", post_back)
+  defp put_custom_vars(%{private: %{elastic_custom_vars: custom_vars}} = map) do
+    custom_vars
+    |> Enum.map(&custom_var(&1))
+    |> Enum.reject(&is_nil/1)
+    |> Enum.reduce(%{}, fn var, acc -> Map.merge(acc, var) end)
+    |> Map.merge(map)
+  end
+
+  defp put_custom_vars(map), do: map
+
+  defp custom_var({:attachments, attachments}), do: %{"attachments" => attachments}
```

Comment:
Change these from `defp custom_var({:attachments, attachments}), do: %{"attachments" => attachments}` to `defp custom_var({:attachments, attachments}), do: {"attachments", attachments}` and the code mentioned above *definitely* becomes `Enum.into/2`. I’m building API transformation this way in my current project all the time.

You may also want to change this specific example to support the API parameter specifically:

```elixir
defp custom_var({:attachments, attachments}), do: {"attachments", List.wrap(attachments)}
```

That assumes that when the Elastic API says "List of strings", they mean an actual JSON list. If it’s not, then we need to understand what it might be (that could be come `Enum.join(List.wrap(attachments), ";")` if it’s semi-colon separated strings, for example).

Reply:
I'll look into what Elastic actually means by "List of Strings".  They use the same declaration for a number of the other fields that are being set already and are just a single string separated by a comma or semicolon (i.e. `msgTo`)